src/main/java/algorithms/easy/binarytreepreordertraversal/BinaryTreePreorderTraversalStack.java

```diff
@@ -0,0 +1,35 @@
+package algorithms.easy.binarytreepreordertraversal;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Stack;
+
+import algorithms.datastructures.TreeNode;
+
+public class BinaryTreePreorderTraversalStack {
+
+	public List<Integer> preorderTraversal(TreeNode root) {
+		if (root == null) {
+			return Collections.emptyList();
+		}
+
+		List<Integer> preorder = new ArrayList<>();
+
+		Stack<TreeNode> stack = new Stack<>();
+		stack.push(root);
+
+		while (!stack.isEmpty()) {
+			TreeNode node = stack.pop();
+			preorder.add(node.val);
+			if (node.right != null) {
+				stack.push(node.right);
+			}
+			if (node.left != null) {
+				stack.push(node.left);
+			}
```

Comment:
```suggestion
            if(node == null) continue;
            preorder.add(node.val);
            stack.push(node.right);
	      stack.push(node.left);
```

Reply:
I think adding null values to the stack will be less efficient than one extra null-check.